### PR TITLE
Move cloud src code

### PR DESCRIPTION
**Pull request by yinweisu**
* Move cloud src code from `autogluon` to `autogluon-cloud`
* This PR move the code unchanged. Any changes needed will be sent in follow-up PRs.
* CI code not moved yet as it will involve changes too

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+spec.loader.exec_module(ag)
+###########################
+
+version = "0.1"
+version = ag.update_version(version, use_file_if_exists=False, create_file=True)
+
+submodule = "cloud"
+install_requires = [
+    # version ranges added in ag.get_dependency_version_ranges()
+    "autogluon.common<0.7",
```
Comment (Innixma) [inline setup.py:23]: Are you sure we should have an open version range for the lower end here?

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    "boto3",
+    "numpy",
+    "packaging<22",
```
Comment (Innixma) [inline setup.py:26]: Are you sure we should have an open version range for the lower end here?

File: setup.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    "pandas",
+    # TODO: update to the latest after 0.6 container is out
+    "sagemaker>=2.94",
+    "pyarrow>=9.0",
+    "Pillow>=9.3.0",
```
Comment (Innixma) [inline setup.py:31]: Maybe cap to some kind of future major version?

**Review by Innixma — APPROVED** [review]
LGTM!

Couple things that would be good to clarify:
1. What is different from the code in autogluon, if anything
2. How will the unit testing / CI work? (Can be added later too)

Finally, the review comments can be addressed in a follow-up PR to avoid altering the alignment between the code in AG and here.

**Comment by yinweisu** [comment]
@Innixma Will be addressing comments in follow-up PRs. Thanks!

File: setup.cfg [hunk 4]
```diff
@@ -0,0 +1,88 @@
+[pycodestyle]
+max_line_length = 160
```
Comment (tonyhoo) [inline setup.cfg:2]: I assume these are consistent with what we have in AG?

File: setup.py (hunk 1 above)
Comment (tonyhoo) [reply setup.py:23]: +1, we shouldn't cap this unless there is a strong reason for it

File: src/autogluon/__init__.py [hunk 5]
```diff
@@ -0,0 +1 @@
+__import__("pkg_resources").declare_namespace(__name__)
```
Comment (tonyhoo) [inline src/autogluon/__init__.py:1]: Shall we change the package name from `autogluon` to `autogluon-cloud`?

File: src/autogluon/cloud/__init__.py [hunk 6]
```diff
@@ -0,0 +1,5 @@
+from autogluon.common.utils.log_utils import _add_stream_handler
```
Comment (tonyhoo) [inline src/autogluon/cloud/__init__.py:1]: Let's remove the `cloud` sub-package as it is a standalone repo now

File: setup.py (hunk 1 above)
Comment (yinweisu) [reply setup.py:23]: OK. I think we need a lower end here: `>=0.6`, but probably not a upper cap

File: setup.cfg (hunk 4 above)
Comment (Innixma) [reply setup.cfg:2]: Yes, this is the length in tabular

**Pull request merged**